### PR TITLE
Create page for log of provider-impacting changes

### PR DESCRIPTION
**Pull request by ajcvickers**
Issue https://github.com/aspnet/EntityFrameworkCore/issues/11198

### CODE REVIEW

**Comment by ajcvickers** [comment]
@roji @cincuranet @ErikEJ @caleblloyd @divega @AndriySvyryd @smitpatel @maumar @bricelam 
 Feedback please!

File: entity-framework/core/providers/provider-log.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+We are starting this log with changes from 2.1 to 2.2. Priot to 2.1 we used the [`providers-beware`](https://github.com/aspnet/EntityFrameworkCore/labels/providers-beware) and [`providers-fyi`](https://github.com/aspnet/EntityFrameworkCore/labels/providers-fyi) labels on our issues and pull requests.
+
+### 2.1 ---> 2.2
+
+#### Test-only changes
+
+* https://github.com/aspnet/EntityFrameworkCore/pull/12057 - Allow customizable SQL delimeters in tests
+  * Test changes that allow non-strict floating point comparisons in BuiltInDataTypesTestBase
+  * Test changes that allow query tests to be re-used with different SQK delimeters
```
Comment (smitpatel) [inline entity-framework/core/providers/provider-log.md]: SQL?

File: entity-framework/core/providers/provider-log.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+---
```
Comment (smitpatel) [inline entity-framework/core/providers/provider-log.md]: Page needs to be added to toc

File: entity-framework/core/providers/provider-log.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+* https://github.com/aspnet/EntityFrameworkCore/pull/12057 - Allow customizable SQL delimeters in tests
+  * Test changes that allow non-strict floating point comparisons in BuiltInDataTypesTestBase
+  * Test changes that allow query tests to be re-used with different SQK delimeters
+* https://github.com/aspnet/EntityFrameworkCore/pull/12072 - Add DbFuction tests to the relational specification tests
+  * Such that these tests can be run against all database providers
+* https://github.com/aspnet/EntityFrameworkCore/pull/12362 - Async test cleanup
+  * Remove `Wait` calls, unneeded async, and renamed some test methods
+* https://github.com/aspnet/EntityFrameworkCore/pull/12666 - Unify logging test infrastructure
+  * `CreateListLoggerFactory` changed, which will require providers using these tests to react
```
Comment (smitpatel) [inline entity-framework/core/providers/provider-log.md]: Grammar?

**Comment by smitpatel** [comment]
Review URL https://review.docs.microsoft.com/en-us/ef/core/providers/provider-log?branch=ajcvickers-provider-log

File: entity-framework/core/providers/provider-log.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+* https://github.com/aspnet/EntityFrameworkCore/pull/12766 - Renaming navigations in the ComplexNavigations model
+  * Providers using these tests may need to react
+* https://github.com/aspnet/EntityFrameworkCore/pull/12141 - Return the context to the pool instead of disposing in functional tests
+  * This change includes some test refactoring which may require providers to react
+
+
+#### Test and product code changes
+
+* https://github.com/aspnet/EntityFrameworkCore/pull/9561 - Add support for System.Transactions
+  * System.Transactions was not previously supported by .NET Core. Now it is, excluding distributed transactions. These changes add support to the EF project code and also add specification tests for use by other providers.
```
Comment (AndriySvyryd) [inline entity-framework/core/providers/provider-log.md]: This was done in 2.1

File: entity-framework/core/providers/provider-log.md (hunk 2 above)
Comment (ajcvickers) [reply entity-framework/core/providers/provider-log.md]: Will do this separately and also link from "writing a provider" page.

File: entity-framework/core/providers/provider-log.md (hunk 1 above)
Comment (ajcvickers) [reply entity-framework/core/providers/provider-log.md]: SQK: A query language for Project K! ;-)

File: entity-framework/core/providers/provider-log.md (hunk 3 above)
Comment (ajcvickers) [reply entity-framework/core/providers/provider-log.md]: Grammar?

File: entity-framework/core/providers/provider-log.md (hunk 4 above)
Comment (ajcvickers) [reply entity-framework/core/providers/provider-log.md]: Ah. I'll remove. :-) I think @roji had it in his list.

File: entity-framework/core/providers/provider-log.md (hunk 3 above)
Comment (smitpatel) [reply entity-framework/core/providers/provider-log.md]: We added new abstract method `CreateListLoggerFactory`. not "changed". It is somewhat confusing for providers what changed. (may be?)

File: entity-framework/core/providers/provider-log.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+ms.date: 08/08/2018
+ms.assetid: 7CEF496E-A5B0-4F5F-B68E-529609B23EF9
+ms.technology: entity-framework-core
+uid: core/providers/provider-log
+---
+
+# Provider-impacting changes
+
+This page contains links to pull requests made on the EF Core repo that may require authors of other database providers to react. The intention is to provide a starting point for authors of existing third-party database providers when updating their provider to a new version.
+
```
Comment (ErikEJ) [inline entity-framework/core/providers/provider-log.md]: Maybe mention the Compliance tests somewhere (I know they are pre 2.2, but)

File: entity-framework/core/providers/provider-log.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+We are starting this log with changes from 2.1 to 2.2. Priot to 2.1 we used the [`providers-beware`](https://github.com/aspnet/EntityFrameworkCore/labels/providers-beware) and [`providers-fyi`](https://github.com/aspnet/EntityFrameworkCore/labels/providers-fyi) labels on our issues and pull requests.
```
Comment (ErikEJ) [inline entity-framework/core/providers/provider-log.md]: Priot => Prior ?

File: entity-framework/core/providers/provider-log.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+### 2.1 ---> 2.2
+
+#### Test-only changes
+
+* https://github.com/aspnet/EntityFrameworkCore/pull/12057 - Allow customizable SQL delimeters in tests
+  * Test changes that allow non-strict floating point comparisons in BuiltInDataTypesTestBase
+  * Test changes that allow query tests to be re-used with different SQK delimeters
+* https://github.com/aspnet/EntityFrameworkCore/pull/12072 - Add DbFuction tests to the relational specification tests
```
Comment (ErikEJ) [inline entity-framework/core/providers/provider-log.md]: DbFuction => DbFunction

File: entity-framework/core/providers/provider-log.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+
+#### Test and product code changes
+
+* https://github.com/aspnet/EntityFrameworkCore/pull/9561 - Add support for System.Transactions
+  * System.Transactions was not previously supported by .NET Core. Now it is, excluding distributed transactions. These changes add support to the EF project code and also add specification tests for use by other providers.
+* https://github.com/aspnet/EntityFrameworkCore/pull/11978 - Consolidate RelationalTypeMapping.Clone methods
+  * Changes to the RelationalTypeMapping allows for a simplification in derived classes. We don't believe this will be breaking to providers, but providers can take advantage of this change in their derived type mapping classes. See https://github.com/aspnet/EntityFrameworkCore/pull/12109 for the changes make to the providers we ship
+* https://github.com/aspnet/EntityFrameworkCore/pull/12069 - Tagged or named queries
+  * Adds infrastructure for tagging LINQ quieries and having those tags show up as comments in the SQL. This may require providers to react in SQL generation.
```
Comment (roji) [inline entity-framework/core/providers/provider-log.md]: quieries -> queries

File: entity-framework/core/providers/provider-log.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+* https://github.com/aspnet/EntityFrameworkCore/pull/12141 - Return the context to the pool instead of disposing in functional tests
+  * This change includes some test refactoring which may require providers to react
+
+
+#### Test and product code changes
+
+* https://github.com/aspnet/EntityFrameworkCore/pull/9561 - Add support for System.Transactions
+  * System.Transactions was not previously supported by .NET Core. Now it is, excluding distributed transactions. These changes add support to the EF project code and also add specification tests for use by other providers.
+* https://github.com/aspnet/EntityFrameworkCore/pull/11978 - Consolidate RelationalTypeMapping.Clone methods
+  * Changes to the RelationalTypeMapping allows for a simplification in derived classes. We don't believe this will be breaking to providers, but providers can take advantage of this change in their derived type mapping classes. See https://github.com/aspnet/EntityFrameworkCore/pull/12109 for the changes make to the providers we ship
```
Comment (ErikEJ) [inline entity-framework/core/providers/provider-log.md]: make => made

**Comment by ErikEJ** [comment]
This is very useful, thanks!

File: entity-framework/core/providers/provider-log.md (hunk 4 above)
Comment (roji) [reply entity-framework/core/providers/provider-log.md]: Hey, don't blame me...

**Comment by roji** [comment]
This will go a long way to help keep up with EF Core changes... thanks!

**Comment by roji** [comment]
This will go a long way to help keep up with EF Core changes... thanks!

File: entity-framework/core/providers/provider-log.md (hunk 5 above)
Comment (ajcvickers) [reply entity-framework/core/providers/provider-log.md]: I think this should go into the "writing a provider" piece, after which I mad add another link from here.

**Pull request merged**